### PR TITLE
V1 - fastfeedback of new P2P

### PR DESCRIPTION
**Pull request by withnale**
* Includes extended test and prod but only fastfeedback is ready for rollout

### CODE REVIEW

File: .github/workflows/p2p-promote-image.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      REGION: ${{ inputs.region }}
+      REGISTRY: ${{ inputs.region }}-docker.pkg.dev/${{ vars.PROJECT_ID }}/tenant
+      SERVICE_ACCOUNT: p2p-${{ vars.TENANT_NAME }}@${{ vars.PROJECT_ID }}.iam.gserviceaccount.com
+      SEMVER: ${{ inputs.semver }}
+      SOURCE_GITHUB_ENV: ${{ fromJson(inputs.source_matrix).include[0]['deploy_env'] }}
+      SOURCE_REGISTRY: ${{ needs.lookup.outputs.source_registry }}
+      SOURCE_SERVICE_ACCOUNT: ${{ needs.lookup.outputs.source_service_account }}
+      SOURCE_WORKLOAD_IDENTITY_PROVIDER: ${{ needs.lookup.outputs.source_workflow_identity_provider }}
+      TAG_VERSION: ${{ inputs.image_tag }}
+      TENANT_NAME: ${{ vars.TENANT_NAME }}
```
Comment (chbatey) [inline .github/workflows/p2p-promote-image.yaml:87]: I am not sure i like this, can they just pass it in?

File: .github/workflows/p2p-workflow-version.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          # fetch-tags is currently broken so we need fetch them manually in the
+          # next step
+          fetch-tags: true
+
+      - name: Get version
+        id: version
+        shell: bash
+        run: |
+          git fetch --prune --unshallow --tags
```
Comment (chbatey) [inline .github/workflows/p2p-workflow-version.yaml:39]: this doesn't actually create the tag, so we can decide if we want that later

File: p2penvtool/p2penvtool.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/env python3
```
Comment (chbatey) [inline p2penvtool/p2penvtool.py:1]: I am not reviewing this, checking it in for reference

File: .github/workflows/p2p-workflow-fastfeedback.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
+
+on:
+  workflow_call:
+    inputs:
+      source:
+        required: false
```
Comment (tmcalves) [inline .github/workflows/p2p-workflow-fastfeedback.yaml:6]: this is missing the secret env_vars

File: .github/workflows/p2p-execute-command.yaml [hunk 5]
```diff
@@ -118,14 +109,6 @@ jobs:
           username: oauth2accesstoken
           password: ${{ steps.auth.outputs.access_token }}
 
-      - name: Decode environment variables
```
Comment (tmcalves) [inline .github/workflows/p2p-execute-command.yaml:121]: why was this removed?

File: .github/workflows/internal-cd.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: CD
+
+on:
+  push:
+    branches:
+      - main
+      - p2p-testing
```
Comment (chbatey) [inline .github/workflows/internal-cd.yaml:7]: to be removed

**Review by tmcalves — APPROVED** [review]

**Comment by tmcalves** [comment]
This has been tested across different apps, merging
